apps/transport/lib/http/utils.ex

```diff
@@ -2,10 +2,71 @@ defmodule Transport.Http.Utils do
   @moduledoc """
   Useful functions to work with http requests
   """
+  require Logger
 
-  def location_header(headers) do
-    for {key, value} <- headers, String.downcase(key) == "location" do
+  def location_header(headers), do: header_value(headers, "location")
+
+  def header_value(headers, header) do
+    for {key, value} <- headers, String.downcase(key) == String.downcase(header) do
       value
     end
   end
+
+  @doc """
+  Takes a response body and response headers and try to convert the body to UTF-8.
+  It looks at the charset defined in the `content-type` header.
+
+  Inspired by https://bernheisel.com/blog/httpoison-and-decompression
+  """
+  def reencode_body_to_utf8(body, headers) do
+    headers
+    |> header_value("content-type")
+    |> parse_charset()
+    |> reencode_body(body)
+  end
+
+  defp parse_charset([content_type]), do: parse_charset(content_type)
+
+  defp parse_charset(content_type) when is_binary(content_type) do
+    case Plug.Conn.Utils.content_type(content_type) do
+      {:ok, _, _, %{"charset" => charset}} ->
+        # Case insensitive and ignore dashes
+        cond do
+          charset =~ ~r/utf-?8/i -> :utf8
```

Comment:
Je me suis demandé ce qui amène ce besoin de flexibilité par rapport à :
- `~r/\Autf-8\z/i`
- ou même une comparaison en chaîne direct lowercase

Ça marche très bien en l'état, c'est de la curiosité et/ou je me demande si plus tard ça ne peut pas poser problème éventuellement.

Reply:
J'ai repris ce qui est dans l'article de blog en ajoutant la partie case insensitive, qui est requis pour le test "real world" INEO et la RFC qui dit que charset doit être case insensitive. `-?` permet de laisser passer un potentiel oubli de dash, genre `UTF8` ou `utf8`.

Reply:
On pourrait ajouter du start of string et end of string !

Reply:
@AntoineAugusti ok ça me va très bien le côté real world ; peut-être qu'on peut du coup mettre une note en commentaire avec l'article ? Ça donne du contexte et ça expliquera la souplesse, pour éviter qu'un mainteneur se pose trop de questions là dessus. Ca me paraît tout à fait légitime du coup de garder ça comme ça avec l'explication !

Reply:
Je pensais que le commentaire dans le code serait suffisant, j'ai mis

> # Case insensitive and ignore dashes

juste au-dessus